native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/HttpClientChannelInitializer.java

```diff
@@ -360,6 +360,9 @@ protected void handshakeFailure(ChannelHandlerContext ctx, Throwable cause) {
         public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
             if (ctx != null && ctx.channel().isActive()) {
```

Comment:
```suggestion
            if (ctx != null) {
                if (ctx.channel().isActive()) {
                    ctx.writeAndFlush(Unpooled.EMPTY_BUFFER).addListener(ChannelFutureListener.CLOSE);
                } else {
                    connectionAvailabilityFuture.notifyFailure(cause);
                    ctx.close();
                }
            }
```

Reply:
Can avoid a redundant null check

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-http/pull/563/commits/154b8e47df3008e4fefce223b39395379a79ea29